src/routes/session/session-get.ts

```diff
@@ -5,10 +5,15 @@ export const SessionGet = (req: Request, res: Response, next: NextFunction) => {
 		return next(new Error("Missing redirect url for session.  Needs to be in format `/session/:redirectUrl`"));
 	}
 
+	let url = req.params[0];
+	if (req.query.baseUrl) {
+		url += encodeURIComponent("?baseUrl=" + req.query.baseUrl);
+	}
+
```

Comment:
This is being used at mutiple places, have a generic logic to pass on query parameters

Reply:
The query parameter is being passed in multiple places, not sure how to make this part generic. Can you please ellaborate a lil bit?

Reply:
Here we're relying on `baseUrl` is being passed as query parameter. Have to write a similar kind of code for another query parametre if added in future. So my observation is to write generic code that pass on all query parameters to the targeted URL. Hope that makes sense.

Reply:
Created a generic method `createUrlWithQueryString` with test cases.

Reply:
You should not pass the query parameter name. It should read all the query parameters available and passed on to the targeted URL.

Reply:
Query parameter name removed. Updated the logic to read all the query parameters